doc/reference.rst

```diff
@@ -338,6 +338,148 @@ use-dependency-links, default: true
 versions, default 'versions'
   The name of a section that contains :ref:`version pins <pinned-versions>`.
 
+Configuration file syntax
+=========================
+
+Buildout configurations use an `INI file format
+<https://en.wikipedia.org/wiki/INI_file>`_.
+
+A configuration is a collection of named sections containing named
+options.
+
+Section names
+-------------
+
+A section begins with a section and and, optionally, a condition in
+square braces (``[`` and ``]``).
+
+A name can consist of any characters other than whitespace, square
+braces, curly braced (``{`` or ``}``), point signs (``#``), colons
+(``:``) or semi-colons (``;``).  The name may be surrounded by leading
+and trailing whitespace, which is ignored.
+
+An optional condition is separated from the name by a colon and is a
+Python expression. It may not contain a pound sign or semi-colon.  See
+the section on :ref:`conditional sections <conditional-sections>` for
+an example and more details.
+
+A comment, preceded by a pound sign or semicolon may follow the
+section name, as in:
+
+.. code-block:: ini
+
+   [buildout] # This is the buildout section
+
+.. -> header
+
+Options
+-------
+
+Options are specified with an option name followed by an equal sign
+and a value:
+
+.. code-block:: ini
+
+   parts = py
+
+.. -> option
+
+    >>> import six
+    >>> import zc.buildout.configparser
+    >>> def parse(s):
+    ...     return zc.buildout.configparser.parse(six.StringIO(s), 'test')
+    >>> from pprint import pprint
+    >>> pprint(parse(header + option))
+    {'buildout': {'parts': 'py'}}
+
+Option names may have any characters other than whitespace, square
+braces, curly braces equal signs or colons.  There may be and usually
+is whitespace between the name and the equal sign and the name and
+equal sign must be on the same line.
+
+Option values may contain any characters. A consequence of this is
+that there can't be comments in option values.
+
+Option values may be continued on multiple lines, and may contain blank lines:
+
+.. code-block:: ini
+
+   parts = py
+
+           test
+
+.. -> option
+
+Whitespace in option values
+___________________________
+
+Trailing whitespace is stripped from each line in an option value.
+Leading and trailing blank lines are stripped from option values.
+
+Handling of leading whitespace and blank lines internal to values
+depend on whether their is data on the first line containing the
+option name.
+
+If there is data on the first line
+  leading whitespace is stripped and blank lines are omitted.
+
+  The
+  resulting option value in the example above is:
+
+  .. code-block:: ini
+
+        py
+        test
+
+  .. -> val
+
+      >>> eq(parse(header + option)['buildout']['parts'] + '\n', val)
+
+If there isn't data on the first line
+  Internal blank lines are retained and common leading white space is stripped.
```

Comment:
More wrapping oddities.  Perhaps this is intentional and has some effect I'm not seeing.